### PR TITLE
Adding and Fixing Functionality For All 26 Cards

### PR DESCRIPTION
**Pull request by dannyallover**
All cards now functional, and all cards except 2nd edition cards, correctly mimic Matt Fisher's (@techmatt) Provincial. We need not worry about 2nd edition cards for now, since Provincial does not include 2nd edition.

### CODE REVIEW

**Comment by dannyallover** [comment]
PTAL @aronsar @benzyx

File: domray/env/domrl/agents/provincial_agent.py [hunk 1]
```diff
@@ -75,103 +89,76 @@ def Sort_List_Of_Tuples(tup):
 
 # helper to check if a card is in their hand
 def hand_contains(state_view, card_name):
-    if card_name in state_view.player.hand:
-        return True
+    for card in state_view.player.hand:
+        if card == card_name:
+            return True
```
Comment (aronsar) [inline domray/env/domrl/agents/provincial_agent.py]: This change isn't necessary; the original two lines is a more pythonic way of doing the same thing.

File: domray/env/domrl/engine/cards/base.py [hunk 2]
```diff
@@ -685,11 +687,17 @@ def condition(self, event):
         return event.event_type == log.EventType.PLAY and event.card.is_type(CardType.ATTACK)
 
     def apply(self, event: log.Event, state):
+        # print("now gets here 3")
         for opp in state.other_players(event.player):
-            if Moat in opp.hand:
+            # print("now gets here 4")
+            # print([card.name for card in opp.hand])
+            opp_hand_card_names = [card.name for card in opp.hand]
+            # print('Moat' in opp_hand_card_names)
```
Comment (aronsar) [inline domray/env/domrl/engine/cards/base.py]: remove commented out print statements

**Review by aronsar — CHANGES_REQUESTED** [review]
Left two comments to beautify the code, but they're small things. I'll pull the changes once they're merged and continue my testing with the updated version.

**Pull request merged**